### PR TITLE
sys/console/minimal: Fix RTT console compiler error for minimal console

### PR DESCRIPTION
**Pull request by vikrant-proxy**
Revert changes in commit b22aea6433dd2663861ed8c2ea56cc229c6c64dd. Instead rename `console_out` to `console_out_nolock` in `rtt_console.c` per suggestions here: https://github.com/apache/mynewt-core/pull/2649#issuecomment-894658567

cc @vrahane @kasjer

### CODE REVIEW

**Review by vrahane — APPROVED** [review]
Looks fine to me. Awaiting @kasjer’s approval before merging it in.

**Review by kasjer — APPROVED** [review]

**Pull request merged**
